src/error/exporterrors.cpp

```diff
@@ -301,247 +276,88 @@ static ostream & printKDBErrors (ostream & os, parse_t & p)
+		   << "			KEY_VALUE, \"" << p[i]["description"] << "\", KEY_END)," << endl;
+		//		   << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "/severity\","
+		//<<
+		// endl
+		//		   << "			KEY_VALUE, \"" << p[i]["severity"] << "\", KEY_END)," << endl
+		//		   << "		keyNew (\"system/elektra/modules/error/specification/" << p[i]["number"] << "/module\"," <<
+		// endl
+		//		   << "			KEY_VALUE, \"" << p[i]["module"] << "\", KEY_END)," << endl;
 	}
 	os << "		KS_END);" << endl << "}" << endl;
 
-	os << "static inline void elektraTriggerWarnings (int nr, Key *parentKey, const char *message)" << endl
-	   << "{" << endl
-	   << "	switch (nr)" << endl
-	   << "	{" << endl;
+	os << "static inline void elektraTriggerWarnings (const char *nr, Key *parentKey, const char *message)" << endl << "{" << endl;
 	for (size_t i = 1; i < p.size (); ++i)
 	{
-		if (p[i]["unused"] == "yes")
-		{
-			continue;
-		}
-
-		if (p[i]["severity"] != "warning") continue;
-		os << "		case " << i << ": ELEKTRA_ADD_WARNING (" << i << ", parentKey, message);" << endl
-		   << "			break;" << endl;
+		os << "	if(strcmp(nr, \"" << p[i]["number"] << "\") == 0)" << endl << "	{" << endl;
+		os << "		ELEKTRA_ADD_" << p[i]["macro"] << "_WARNING (parentKey, message);" << endl
+		   << "		return;" << endl
+		   << "	}" << endl;
 	}
-	os << "		default: ELEKTRA_ADD_WARNING (45, parentKey, \"in default branch\");" << endl
-	   << "			 break;" << endl
-	   << "	}" << endl
-	   << "}" << endl
-	   << "" << endl;
-	os << "static inline void elektraTriggerError (int nr, Key *parentKey, const char *message)" << endl
-	   << "{" << endl
-	   << "	switch (nr)" << endl
-	   << "	{" << endl;
+	os << "	ELEKTRA_ADD_INTERNAL_WARNING (parentKey, \"Unkown warning code\");" << endl << "}" << endl << "" << endl;
+	os << "static inline void elektraTriggerError (const char *nr, Key *parentKey, const char *message)" << endl << "{" << endl;
 	for (size_t i = 1; i < p.size (); ++i)
 	{
-		if (p[i]["unused"] == "yes")
-		{
-			continue;
-		}
-
-		if (p[i]["severity"] == "warning") continue;
-		os << "		case " << i << ": ELEKTRA_SET_ERROR (" << i << ", parentKey, message);" << endl
-		   << "			break;" << endl;
+		os << "	if(strcmp(nr, \"" << p[i]["number"] << "\") == 0)" << endl << "	{" << endl;
+		os << "		ELEKTRA_SET_" << p[i]["macro"] << "_ERROR (parentKey, message);" << endl
+		   << "		return;" << endl
+		   << "	}" << endl;
 	}
-	os << "		default: ELEKTRA_SET_ERROR (44, parentKey, \"in default branch\");" << endl
-	   << "			 break;" << endl
-	   << "	}" << endl
-	   << "}" << endl;
-
+	os << "	ELEKTRA_SET_INTERNAL_ERROR (parentKey, \"Unkown error code\");" << endl << "}" << endl;
 
 	os << "#endif" << endl;
 	return os;
 }
 
-static ostream & printCreatorSignature (ostream & os, map<string, string> & data)
+static string macroCaseToPascalCase (const string & s)
```

Comment:
What is PascalCase? You mean CamelCase?

Reply:
https://stackoverflow.com/questions/41768733/camel-case-and-pascal-case-mistake